### ISSUE TITLE
Application closes itself during running

### ISSUE THREAD

**Issue by gpuaux** [khertan/ForRunners#64]
During a running, (after 35 min), GPS signal was lost. After that, I didn't hear the vocal message occuring each 5 min. When my session was done, the application was no longer running on the phone.
I opened the application after that, but no session was recorded.
It was the first time use of this application. 

The phone is a XIAOMI Redmi 4A, with MIUI Global 8.5.1.0 Stable (Android version 6.0.1 MMB29M). 
Application installed from FDroid.

Is there some debug files I can give for this issue ?

**Comment by brvier** [comment]
I already got similar report on Xiaomi device. It s look like the ‘Battery Saver’ feature implemented by Xiaomi didn’t respect background prefs of apps and kill them.

http://nine-faq.9folders.com/articles/8771-how-to-turn-off-background-restriction-on-the-xiaomi-devices

Benoît HERVIER - http://rvier.fr

**Comment by gpuaux** [comment]
In fact I already added the ForRunners application in the list of applications with no restriction for battery. So this is not the reason for the application to close.

After the running session, because of the application closing, I decided then to disable entirely the battery usage restriction. My phone will no longer restrict all the applications. For my next running session, I will then test that the application is running well.

Thanks a lot for developing this app :)

**Comment by brvier** [comment]
Ouch so ... it should not be closed ... this require more investigation.

Anyway I change many things in planned release but I m still tracking one bug before releasing that appear only on some device.

Did you have developer mode activated ?

Benoît HERVIER - http://rvier.fr

**Comment by gpuaux** [comment]
Yes, I have the developer mode activated. Only for usb debugging to be able to share files with my computer. I have no experience on android dev, but if I can help by doing something...

Grégory

**Comment by gpuaux** [comment]
Hello
I can give maybe more information.
I ran today trying using another sport tracking software. With two different softwares running simultaneously plus spotify, they all stopped during my running session.
I don't know if they stopped simultaneously or not (at a moment the music stopped, I checked the smartphone et saw none of the softwares running).

So, maybe the problem is not specific to forrunners but maybe it happen when some softwares uses intensively the GPS. I will try soon with a hikking tracking software and give you some feedback.
If you need that I log some files with forrunner don't hesitate to ask for.

Best

Grégory

**Comment by brvier** [comment]
This could be close by ressource management of your device, but yes, logs are always welcome
